src/IFrameNavigator.ts

```diff
@@ -102,10 +103,11 @@ export default class IFrameNavigator implements Navigator {
     private annotator: Annotator | null;
     private settings: BookSettings;
     private eventHandler: EventHandler;
+    private upUrl: URL | null;
+    private upLabel: string | null;
```

Comment:
Since you need either both of these or neither of these (I assume), I might do the following instead—it would help you eliminate nonsensical states:

```typescript
private up: { url: URL, label: string} | null;
```

If you do this, you'll need to change the constructor too, of course.